src/BucketTesting/Logging/StreamBucketLogger.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\BucketTesting\Logging;
+
+use DateTime;
+use WMDE\Fundraising\Frontend\BucketTesting\Bucket;
+
+class StreamBucketLogger implements BucketLogger {
+
+	private $stream;
+	private $dateFormat = DateTime::RFC3339_EXTENDED;
+
+	/**
+	 * @param resource $stream
+	 */
+	public function __construct( $stream ) {
+		$this->stream = $stream;
+	}
+
+	public function writeEvent( LoggingEvent $event, Bucket ...$buckets ) {
+		fwrite(
+			$this->stream,
+			json_encode( [
+				'date' => date( $this->dateFormat ),
```

Comment:
This is a global function that relies on global state that gets changed by the system. As you noticed this causes issues when trying to test this code. What about creating a TimeThinghy service? That way this code no longer binds to that changing global state and also no longer needs to know about the time format.

Reply:
I will make a PR into this one with a SuchSuperTimeThinghy